flocker/acceptance/testtools.py

```diff
@@ -203,6 +203,61 @@ def _clean_node(test_case, node):
         pass
 
 
+def get_volume_backend(test_case):
+    """
+    Get the volume backend the acceptance tests are running as.
+
+    :param test_case: The ``TestCase`` running this unit test.
+
+    :raise SkipTest: if the backend is specified.
+    """
+    backend = environ.get("FLOCKER_ACCEPTANCE_VOLUME_BACKEND")
+    if backend is None:
+        raise SkipTest(
+            "Set acceptance testing volume backend using the " +
+            "FLOCKER_ACCEPTANCE_VOLUME_BACKEND environment variable.")
+    return backend
+
+
+def require_backend(supported, reason):
+    """
+    Decorator that skips a test if the volume backend doesn't support
+    the operations required by the test.
+
+    :param supported: List of supported volume backends for this test.
+    :param reason: The reason the backend isn't supported.
+    """
+    def decorator(test_method):
+        """
+        :param test_method: The test method that will be called when the
+            cluster is available and which will be supplied with the
+            ``cluster``keyword argument.
+        """
+        def call_test_method_with_cluster(cluster, test_case, args, kwargs):
+            kwargs['cluster'] = cluster
+
+        @wraps(test_method)
+        def wrapper(test_case, *args, **kwargs):
+            backend = get_volume_backend(test_case)
+
+            if backend not in supported:
+                raise SkipTest(
+                    "Backend not supported: {backend} ({reason}). "
+                    "Supported backends: {supported}.".format(
+                        backend=backend,
+                        reason=reason,
+                        supported=', '.join(supported),
+                    )
+                )
+            return test_method(test_case, *args, **kwargs)
+        return wrapper
+    return decorator
+
+require_moving_backend = require_backend(
+    supported=[b"zfs", b"openstack", b"ebs"],
```

Comment:
This is going to hamper third-party testing.  On the one hand, I'd really like to not have to think about that right now.  On the other hand, I don't know if we'll be able to get away with that.  Rather than agonize over that, maybe it'd be easier just to suggest inverting this and blacklisting `loopback` instead of whitelisting these three.